weave/trace/op_extensions/accumulator.py

```diff
@@ -15,6 +15,11 @@
 
 from weave.trace.op import FinishCallbackType, Op
 
+try:
```

Comment:
I have an idea to make changes to this file much less invasive. For context, there are a lot of changes in this file just to manage the passing of around of the `skip_last` flag and managing what to do in such circumstances. This feels like we might be trying to do too much with this one function. Here is alternative proposal:

1. Change the `def add_accumulator` method's `accumulator` arg to `make_accumulator` which is a function that accepts the inputs and returns an accumulator function. Note: this will require changing a few other integrations to wrap their accumulator with a lambda... but it will be worth it i think.
2. From there, in your `openai_sdk` integration, you can make your accumulator raise a `StopIteration` exception when: a) you would normally skip last and b) you are processing a usage payload (which is always the last one).

Doing this will ensure that the functionality of this module is not overloaded.

Reply:
> Change the def add_accumulator method's accumulator arg to make_accumulator which is a function that accepts the inputs and returns an accumulator function.

This is a nice idea to remove the clutter of checking for one specific case (integration) in the `accumulator.py` file. I implemented it and it works fine.

> From there, in your openai_sdk integration, you can make your accumulator raise a StopIteration exception when: a) you would normally skip last and b) you are processing a usage payload (which is always the last one).

Say I raise `StopIteration` when `skip_last` is True and it the last item in the iterator with `usage` info. It will not return the said value (with usage info) to the user. I achieved this however, since we are not returning the final `acc` (that has usage info), it is not logged to weave. Need to handle this in an ugly way maybe.

Reply:
Ignore, I was able to do it in an elegant way. TIL `StopIteration(value)` is a thing.

Reply:
You rock <3